packages/widget/src/components/common/buttons/connect-wallet.ts

```diff
@@ -62,49 +61,124 @@ export class ConnectWalletButton extends BaseComponent {
     return !!this.wallets.evmWallet || !!this.wallets.substrateWallet;
   }
 
-  render(): HTMLTemplateResult {
-    const evmWallet = this.wallets.evmWallet;
+  private handleSubstrateAccountSelected = (
+    option: DropdownOption<Account>
+  ): void => {
+    this.dispatchEvent(
+      new WalletUpdateEvent({
+        substrateWallet: {
+          signer: this.wallets.substrateWallet!.signer,
+          signerAddress: option.value.address,
+          accounts: this.wallets.substrateWallet?.accounts ?? []
+        }
+      })
+    );
+  };
+
+  private renderDisconnectSubstrateButton(): HTMLTemplateResult | undefined {
```

Comment:
Because for the substrate it's a part of the dropdown list as the last element and it needs special styling

Reply:
Create SubstrateAccountSelector component then. Although, it probably makes sense to ignore design there and leave disconnect button as is right now

Reply:
agreed most (if not all) of substrate logic should be in a separate component (instead of having EVM logic in one place and Substrate logic in connect button)

Reply:
It makes sense to me as well, the reason I left it this way is my past experiences with the props drilling in Lit and especially styles customization. Let me try to refactor it and see if It makes the life easier.